addons/capetrader/capetrader.lua

```diff
@@ -0,0 +1,495 @@
+        elseif currentlyAugmenting then
+            windower.add_to_chat(123, 'You are currently still augmenting a cape, please wait until the process finishes.')
+        end
+    elseif cmd == 'list' or cmd == 'l' then
+        printAugList()
+    elseif cmd == 'help' or cmd == 'h' then
+        printHelp()
+    elseif cmd == 'unload' or cmd == 'u' then
+        windower.send_command('lua unload ' .. _addon.name)
+    elseif cmd == 'reload' or cmd == 'r' then
+        windower.send_command('lua reload ' .. _addon.name)
+    else
+        windower.add_to_chat(123, 'You entered an unknown command, enter //ct help if you forget your commands.')
+    end
+end)
+
+function getCapeIndex()
+    for itemIndex, item in pairs(inventory) do
+        if item.id == currentCape.id then
+            return itemIndex
+        end
+    end
+end
+
+function getAugItemIndex()
+    for itemIndex, item in pairs(inventory) do
+        if item.id == pathItem.id then
+            return itemIndex
+        end
+    end
+end
+
+function getItem(itemName)
+    for k, item in pairs(res.items) do
+        if string.lower(item.en) == string.lower(itemName) then
+            return item
+        end
+    end
+end
+
+function buildTrade()
+    local packet = packets.new('outgoing', 0x036, {
+        ["Target"] = gorpaID,
+        ["Target Index"] = gorpaTargetIndex,
+        ["Item Count 1"] = 1,
+        ["Item Count 2"] = 1,
+        ["Item Index 1"] = capeIndex,
+        ["Item Index 2"] = augmentItemIndex,
+        ["Number of Items"] = 2
+    })
+    packets.inject(packet)
+end
+
+windower.register_event('incoming chunk', function(id, data, modified, injected, blocked)
+    if id == 0x00B or id == 0x00A then
+        zoneHasLoaded = false --NOTE: This idea was taken from Zohno's findall addon.
+    end
+
+    if id == 0x034 or id == 0x032 then
+        if currentlyAugmenting and playerIndex then
+
+            injectAugmentConfirmationPackets()
+
+            timesAugmentedCount = timesAugmentedCount + 1
+            if timesAugmentedCount <= tonumber(numberOfTimesToAugment) then
+                windower.add_to_chat(158, timesAugmentedCount - 1 .. '/' .. numberOfTimesToAugment .. ' augments completed.')
+                tradeReady = true
+            else
+                capeHasBeenPrepared = false
+                currentlyAugmenting = false
+                playerIndex = nil
+                currentCape = nil
+                pathItem = nil
+                tradeReady = false
+                functions.schedule(sendCompletedMessage, 2)
+            end
+
+            return true
+        end
+    end
+
+    if id == 0x01D then
+        if not zoneHasLoaded then
+            zoneHasLoaded = true
+        end
+        if tradeReady then
+            tradeReady = false
+
+            if not string.contains(string.lower(pathItem.en), 'dye') then
```

Comment:
This may just be your preference, but this is more succintly typed when using the object syntax:
```lua
            if not pathItem.en:lower():contains('dye') then
```

This can be applied to many cases throughout this file, and with functions too, such as in the next line:
```lua
                startAugmentingCape:schedule(dustSapThreadTradeDelay, numberOfTimesToAugment - timesAugmentedCount + 1, false)
```